### ISSUE TITLE
Empty application path

### ISSUE THREAD

**Issue by Zinggi** [AntiMicroX/antimicrox#369]
Sometimes antimicrox couldn't give me the correct application path for the running app, needed for auto profiles to work.

This happens when the pid of the running application is too high.

This patch fixes it:

```cpp
diff --git a/src/x11extras.cpp b/src/x11extras.cpp
index 930a8707..9fa156e9 100644
--- a/src/x11extras.cpp
+++ b/src/x11extras.cpp
@@ -268,7 +268,8 @@ int X11Extras::getApplicationPid(Window window)
 
         if ((status == 0) && (prop != nullptr))
         {
-            pid = prop[1] << 8;
+            pid = prop[2] << 16;
+            pid += prop[1] << 8;
             pid += prop[0];
             XFree(prop);
         }
```

`prop[2]` contains a `1` in this case. 
I'm sure there is a better way to find the correct pid, but I had no time to research this further.

**Comment by pktiuk** [comment]
Hi @Zinggi 

Thanks for your report and potential fix :)  
You saved me a lot of time. I will investigate it further.

**Comment by Zinggi** [comment]
I'm glad it was useful :)

But I probably wouldn't use this fix directly. There probably is a better way to do it.
I was just hacking on something else and this fix worked, so I could get on with it.
Maybe on some systems the third element of that array isn't part of the pid? Maybe thi now breaks on some other configurations? Point is, I have no idea what I'm doing, just got lucky.

**Comment by pktiuk** [comment]
I checked it.  
Max PID for Linux-bases systems is `4194304` (`cat /proc/sys/kernel/pid_max`)  
After this fix, we should cover PIDs up to `16777215`  

Moreover, I checked what is inside returned array and there were no problems, because further values were 0 (prop[3], prop[5], prop[6])
